site-src/geps/gep-726.md

```diff
@@ -0,0 +1,319 @@
+implementations.
+
+Although not all of these implementations directly support prefix rewrites or
+redirects, the ones that don't include regular expression support which can be
+used to implement prefix rewrites and redirects.
+
+Note: This section intentionally excludes the redirect capabilities already
+contained in the API.
+
+### Envoy
+Envoy supports the following relevant capabilities
+([reference](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto)):
+
+* path_redirect (redirect only)
+* prefix_rewrite (redirect and forwarding)
+* regex_rewrite (redirect and forwarding)
+* host_rewrite_literal (forwarding only)
+* strip_query (redirect only)
+
+Note that path rewrite relies on the prefix match for the route, there is not
+a way to differentiate between the prefix used for matching and rewriting.
+
+### Google Cloud
+Google Cloud URL Maps support the following relevant capabilities
+([reference](https://cloud.google.com/compute/docs/reference/rest/v1/urlMaps)):
+
+* pathPrefixRewrite (forwarding only)
+* hostRewrite (forwarding only)
+* pathRedirect (redirect only)
+* prefixRedirect (redirect only)
+* stripQuery (redirect only)
+
+Note that path rewrite relies on the prefix match for the route, there is not
+a way to differentiate between the prefix used for matching and rewriting.
+
+### HAProxy
+HAProxy supports the following relevant capabilities
+([reference](https://cbonte.github.io/haproxy-dconv/2.5/configuration.html)):
+
+* http-request set-path (advanced path rewrite capabilities)
+* http-request replace-path (rewrites entire path)
+* http-request replace-pathq (rewrites entire path + query string)
+* http-request replace-uri (URI rewrite based on input regex)
+* redirect location (advanced redirect capabilities)
+
+### NGINX
+The NGINX rewrite module contains the following relevant capabilities
+([reference](http://nginx.org/en/docs/http/ngx_http_rewrite_module.html)):
+
+* PCRE regex based rewrites
+* Rewrite directive can be used during forwarding or redirects
+* Rewrite directive can affect host, path, or both
+* Rewrite directive can be chained
+
+## Future Extension
+There are two relatively common types of path rewrite/redirect that are not
+covered by this proposal:
+
+1. Replace a path prefix separate from the match
+2. Replace with a Regular Expression substitution
+
+Both of the following can be represented by adding a new field new types. For
+example, this config would result in a request to `/foo/baz` to be rewritten to
+`/bar/baz`:
+
+```yaml
+filters:
+- type: RequestRewrite
+  requestRewrite:
+    path:
+      type: ReplacePrefix
+      pattern: /foo
+      substitution: /bar
+```
+
+Similarly, this config would result in a request to `/foo/bar/baz` being
+rewritten to `/foo/other/baz`.
+```yaml
+filters:
+- type: RequestRewrite
+  requestRewrite:
+    path:
+      type: RegularExpression
+      pattern: /foo/(.*)/baz
```

Comment:
isn't "*" enough here ?

Reply:
I think you are thinking of UNIX glob expression vs regular expression.

The dot "." means any character.